### PR TITLE
Faros Destination: dynamically require params based on selected edition

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

Utilize Airbyte UI capabilities to dynamically present configuration options to user based on the selected Faros edition.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/faros-destination/src/index.ts [hunk 1]
```diff
@@ -61,9 +61,9 @@ export enum InvalidRecordStrategy {
   SKIP = 'SKIP',
 }
 
-export enum RunMode {
-  COMMUNITY_EDITION = 'community_edition',
-  FULL_EDITION = 'full_edition',
+export enum Edition {
```
Comment (tovbinm) [inline destinations/faros-destination/src/index.ts:64]: let's do `community` and `cloud`

File: destinations/faros-destination/src/index.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ class FarosDestination extends AirbyteDestination {
         message: e.message,
       });
     }
-    if (config.run_mode === RunMode.COMMUNITY_EDITION) {
+    if (this.edition === Edition.COMMUNITY) {
```
Comment (tovbinm) [inline destinations/faros-destination/src/index.ts:121]: is there enough shared stuff  in `check` function? other wise I would suggest:
```typescript
async check(config: AirbyteConfig): Promise<AirbyteConnectionStatusMessage> {
const edition = config.edition_configs?.edition;
    if (!edition) {
      throw new VError('Faros Edition is not set');
    }
    this.edition = edition;
    if (edition === Edition.COMMUNITY) {
        await checkCommunity(config)
    } else if (edition === Edition.PREMIUM) {
        await checkPremium(config)
    }  else {
        throw new VError(
        `Invalid run mode ${edition}. ` +
          `Possible values are ${Object.values(Edition).join(',')}`
      );
    }
}
```

File: faros-airbyte-cdk/src/utils.ts [hunk 3]
```diff
@@ -14,13 +15,20 @@ export const PACKAGE_VERSION = packageInfo.version;
...
 /** Redact config of all secret values based on the provided specification */
 export function redactConfig(config: AirbyteConfig, spec: AirbyteSpec): string {
-  const props = spec.spec.connectionSpecification?.properties ?? {};
   const paths = [];
-  for (const prop of Object.keys(props)) {
-    if (props[prop]?.airbyte_secret === true) {
-      paths.push(prop);
-    }
-  }
+  traverse(spec.spec.connectionSpecification ?? {}, {
```
Comment (tovbinm) [inline faros-airbyte-cdk/src/utils.ts:19]: does this apply a recursive traverse?

**Review by tovbinm — APPROVED** [review]
a couple of comments, otherwise lgtm!

File: faros-airbyte-cdk/src/utils.ts (hunk 3 above)
Comment (cjwooo) [reply faros-airbyte-cdk/src/utils.ts:19]: Yes

File: destinations/faros-destination/test/temp.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import fs from 'fs';
+import {Mockttp} from 'mockttp';
```
Comment (tovbinm) [inline destinations/faros-destination/test/temp.ts:2]: @cjwooo I would rename this from `temp.ts` to `testing-tools.ts` or something

**Pull request merged**